tests/integration-tests/tests/scheduler_plugin/test_scheduler_plugin.py

```diff
@@ -464,3 +476,60 @@ def _test_sudoer_configuration(command_executor, run_as_user):
     # Test user2 is able to run "ls" command as root
     result = command_executor.run_remote_command(f"sudo su {user2} -c 'sudo -u root ls'")
     assert_that(result.failed).is_false()
+
+
+def _test_update_compute_fleet_status_script(command_executor):
+    """Test update-compute-fllet-status.sh."""
+    result = command_executor.run_remote_command("update-compute-fleet-status.sh --status PROTECTED").stdout
+    assert_that(result).is_empty()
+    _test_get_compute_fleet_status_script(command_executor, "PROTECTED")
+
+
+def _test_get_compute_fleet_status_script(command_executor, expected_status):
+    """Test get-compute-fleet-status.sh."""
+    result = command_executor.run_remote_command("get-compute-fleet-status.sh").stdout
+    status = json.loads(result)
+    assert_that("lastStatusUpdatedTime" in status and "status" in status).is_true()
+    assert_that(status.get("status")).is_equal_to(expected_status)
+
+
+def _test_cluster_update(cluster, before_update_cluster_config, cluster_config):
+    """Test cluster update."""
+    # Command executor
+    command_executor = RemoteCommandExecutor(cluster)
+    # Test cluster configuration before cluster update
+    _test_cluster_config(command_executor, before_update_cluster_config)
+    # Test HeadComputeFleetUpdate handler and get-compute-fleet-status.sh when stop cluster
+    _test_cluster_stop(cluster, command_executor)
+    cluster.update(str(cluster_config), force_update="true")
+    command_executor = RemoteCommandExecutor(cluster)
+    # Test HeadComputeFleetUpdate handler and get-compute-fleet-status.sh when start cluster
+    _test_cluster_start(cluster, command_executor)
+    command_executor = RemoteCommandExecutor(cluster)
+    # Test cluster configuration after cluster update
+    _test_cluster_config(command_executor, cluster_config)
```

Comment:
we should also test that the content of PCLUSTER_CLUSTER_CONFIG_OLD is correct

Reply:
Added